RoundsWithFriends/GameModes/DeathmatchHandler.cs

```diff
@@ -16,7 +18,8 @@ public DeathmatchHandler() : base("Deathmatch")
         {
             this.Settings = new GameSettings() {
                 { "pointsToWinRound", 2 },
-                { "roundsToWinGame", 3 }
+                { "roundsToWinGame", 3 },
+                { "allowTeams", false }
             };
```

Comment:
Settings are meant as a set of configurable variables that could be changed in the lobby before starting the game. It's obviously missing UI still, but I want to keep the idea intact. I offer two paths:
1. Get out of my basement: Remove the "allowTeams" parameter from handler settings and move it somewhere else (I don't really care where). It's maybe generic enough to make part of the handler interface, but I'll leave it to your consideration.
2. Stay in the basement: Combine Deathmatch and Team Deathmatch (essentially the only difference is the settings) and implement UI for changing handler settings.

Reply:
I sorta did both. Why did I do both.

I will be moving the `allowTeams`, and `playersRequiredToStartGame` over to `IGameModeHandler` (but not `maxPlayers` or `maxTeams`) but I also combined TDM and DM into a single abstracted gamemode which other mods can (and already do 😉 inherit from)